modules/virt-using-wasp-agent-to-configure-higher-vm-workload-density.adoc

```diff
@@ -0,0 +1,286 @@
+      evictionSoftGracePeriod:
+        memory.available: 1m30s
+      failSwapOn: false
+----
+.. Run the following command:
++
+[source,yaml]
+----
+$ oc wait mcp worker --for condition=Updated=True
+----
+. Create a `MachineConfig` object to provision swap as follows:
++
+[source,terminal]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+labels:
+machineconfiguration.openshift.io/role: worker
+name: 90-worker-swap
+spec:
+config:
+ignition:
+version: 3.4.0
+systemd:
+units:
+- contents: |
+[Unit]
+Description=Provision and enable swap
+ConditionFirstBoot=no
+
+          [Service]
+          Type=oneshot
+          Environment=SWAP_SIZE_MB=5000
+          ExecStart=/bin/sh -c "sudo dd if=/dev/zero of=/var/tmp/swapfile count=$SWAP_SIZE_MB bs=1MiB && sudo chmod 600 /var/tmp/swapfile && sudo mkswap /var/tmp/swapfile && sudo swapon /var/tmp/swapfile && free -h"
+
+          [Install]
+          RequiredBy=kubelet-dependencies.target
+        enabled: true
+        name: swap-provision.service
+----
++
+To have enough swap for the worst-case scenario, make sure to have at least as much swap space provisioned as overcommitted RAM. Calculate the amount of swap space to be provisioned on a node using the following formula:
++
+[source,terminal]
+----
+NODE_SWAP_SPACE = NODE_RAM * (MEMORY_OVER_COMMIT_PERCENT / 100% - 1)
+----
++
+Example:
++
+[source,terminal]
+----
+NODE_SWAP_SPACE = 16 GB * (150% / 100% - 1)
+                = 16 GB * (1.5 - 1)
+                = 16 GB * (0.5)
+                =  8 GB
+----
+. Deploy alerting rules as follows:
++
+[source,terminal]
+----
+apiVersion: monitoring.openshift.io/v1
+kind: AlertingRule
+metadata:
+name: wasp-alerts
+namespace: openshift-monitoring
+spec:
+groups:
+- name: wasp.rules
+rules:
+- alert: NodeSwapping
+annotations:
+description: Node {{ $labels.instance }} is swapping at a rate of {{ printf "%.2f" $value }} MB/s
+runbook_url: https://github.com/openshift-virtualization/wasp-agent/tree/main/runbooks/alerts/NodeSwapping.md
+summary: A node is swapping memory pages
+expr: |
+# In MB/s
+irate(node_memory_SwapFree_bytes{job="node-exporter"}[5m]) / 1024^2 > 0
+for: 1m
+labels:
+severity: critical
+----
+. Configure OpenShift Virtualization to use memory overcommit:
+** Using the {product-title} web console,
++
+or
++
+** As follows:
++
+[source,terminal]
+----
+apiVersion: hco.kubevirt.io/v1beta1
+kind: HyperConverged
+metadata:
+name: kubevirt-hyperconverged
+namespace: openshift-cnv
+spec:
+higherWorkloadDensity:
+memoryOvercommitPercentage: 150
+----
+. Run the following command:
++
+[source,terminal]
+----
+$ oc patch --type=merge \
+  -f <../manifests/hco-set-memory-overcommit.yaml> \
+  --patch-file <../manifests/hco-set-memory-overcommit.yaml>
+----
+
+[NOTE]
+====
+After applying all configurations, the swap feature is fully available only after all `MachineConfigPool` rollouts are complete.
+====
+
+.Verification
+
+. Verify the deployment TBD
```

Comment:
```suggestion
.. Run the following command:
  [source, terminal]
  ----
$  oc rollout status ds wasp-agent -n wasp
  ----
 If deployment was successful you should see "daemon set "wasp-agent" successfully rolled out"
```

Reply:
@enp0s3  Is this the exact message that the user will see, with "wasp-agent" in quotes:

daemon set "wasp-agent" successfully rolled out

Reply:
@sbeskin-redhat Correct.